### PR TITLE
Fixed a bug that may cause TypeError.

### PR DESCRIPTION
**Pull request by Gabrielvon**
Once the data is already numpy.ndarray, it will return NoneType valua which overwrite the rawdata in the other functions.

### CODE REVIEW

**Comment by Gabrielvon** [comment]
fixed a bug that may cause TypeError. to Fixed a bug that may cause TypeError

**Comment by tsurubee** [comment]
@Gabrielvon 
Sorry for my late reply.  And, thank you for a wonderful update !!
I fully agree with your change. 
So, as soon as CI passes through it, it will be imported into master. 
Thank you.

**Comment by Gabrielvon** [comment]
@tsurubee no probs. I wonder if it goes through? What does `All checks have failed` means? It is my first time to pull and request to merge so sorry thatI don't what is the right way to do it.

**Comment by tsurubee** [comment]
@Gabrielvon
Sorry for my late reply.   `All checks have failed` means that not all of the CI tests go through (see [Details of Travis CI result](https://travis-ci.org/tsurubee/banpei/builds/413113882?utm_source=github_status&utm_medium=notification))  
And,  the parts that are in error are as follows
```
Traceback (most recent call last):
  File "/home/travis/build/tsurubee/banpei/tests/test_hotelling.py", line 16, in test_hotelling
    self.assertEqual(expected, len(results))
AssertionError: 2 != 200
```

File: banpei/hotelling.py [hunk 1]
```diff
@@ -22,16 +22,15 @@ def detect(self, data, threshold):
...
         data_abn = [(x - avg)**2 / var for x in data]
 
         # Abnormality determination
-        result = []
-        for (index, x) in enumerate(data_abn):
-            if x > abn_th:
-                result.append((index, data[index]))
+        result = np.full_like(data, np.nan)
+        tf_ind = data_abn > abn_th
+        result[tf_ind] = data[tf_ind]
```
Comment (tsurubee) [inline banpei/hotelling.py]: Originally, since it was implemented so as to return an array of the same size as the input data, a CI error has appeared.  
I am in favor of the change to speed up with ndarray!

File: banpei/utils.py [hunk 2]
```diff
@@ -33,3 +33,43 @@ def power_method(A, iter_num=1):
...
+        roll along height(axis=0): rolling_window(a.transpose(2, 1, 0), 3).transpose(2, 3, 1, 0)
+        roll along width(axis=1): rolling_window(a, 3).transpose(2, 0, 1, 3)
+        roll along depth(axis=2): rolling_window(a.transpose(0, 2, 1), 3).transpose(3, 0, 2, 1)
+    """
+    shape = a.shape[:-1] + (a.shape[-1] - window + 1, window)
+    strides = a.strides + (a.strides[-1],)
+    return np.lib.stride_tricks.as_strided(a, shape=shape, strides=strides)
+
+
+def rolling_window(arr, window, axis=0):
```
Comment (tsurubee) [inline banpei/utils.py]: Is this a method you added to extend SST to multidimensional time series data?
I asked because it exceeded the range of PR title `Fixed a bug that may cause TypeError.`.

File: banpei/hotelling.py (hunk 1 above)
Comment (Gabrielvon) [reply banpei/hotelling.py]: Yes, I made a bit more modification on my version.

File: banpei/utils.py (hunk 2 above)
Comment (Gabrielvon) [reply banpei/utils.py]: My bad. I forgot I made more modification on my version. Sorry about it. Could you just changed the part of bugfix or whatever you want to add to your current version? Otherwise, I can change it on my end. What should I do next?

File: banpei/hotelling.py (hunk 1 above)
Comment (tsurubee) [reply banpei/hotelling.py]: Changing the size of the return value means that a program that is working normally will not work by merging your change.    
You should adjust and modify them so that CI tests pass.

**Comment by tsurubee** [comment]
What I want to tell you is that the scope of PR is not appropriate.

Although you mentioned only TypeError in the above comment, various codes such as speeding up Hotelling theory and multidimensional extension of SST etc. are committed without explanation to actual PR contents.  
In such a case, thinking about the side of the reviewer, I think that it is better to separate the PR appropriately.  

In conclusion what you want to do to merge this PR;
First, changing your code to go through the CI test.
Second, delete the commit that exceeds the scope of the PR title or clip it to another PR.

If what you want to use is different from what I'm thinking, you can fork it and change it as you like, or you can copy it and make it in another form.
Thank you.

**Comment by Gabrielvon** [comment]
I see. I am thinking keep all changes on my forked version since that is more suitable for me. For now, shall we discard this request? I may pull requests step by step based on the scope of changes in the near future.

**Comment by tsurubee** [comment]
Thank you for understanding.  
Many of your changes were great, but you packed too much with PR, I can not merge for master.  

Thank you so much for your PR!

**Comment by Gabrielvon** [comment]
Totally understood. I am pretty new to git so that lots of rule that I am not familiar with. Thanks for guiding me on this operations.
